libs/datamodel/connectors/sql-datamodel-connector/src/cockroach_datamodel_connector.rs

```diff
@@ -0,0 +1,301 @@
+const OID_TYPE_NAME: &str = "Oid";
+const REAL_TYPE_NAME: &str = "Real";
+const DOUBLE_PRECISION_TYPE_NAME: &str = "DoublePrecision";
+const VARCHAR_TYPE_NAME: &str = "VarChar";
+const CHAR_TYPE_NAME: &str = "Char";
+const TEXT_TYPE_NAME: &str = "Text";
+const BYTE_A_TYPE_NAME: &str = "ByteA";
+const TIMESTAMP_TYPE_NAME: &str = "Timestamp";
+const TIMESTAMP_TZ_TYPE_NAME: &str = "Timestamptz";
+const DATE_TYPE_NAME: &str = "Date";
+const TIME_TYPE_NAME: &str = "Time";
+const TIME_TZ_TYPE_NAME: &str = "Timetz";
+const BOOLEAN_TYPE_NAME: &str = "Boolean";
+const BIT_TYPE_NAME: &str = "Bit";
+const VAR_BIT_TYPE_NAME: &str = "VarBit";
+const UUID_TYPE_NAME: &str = "Uuid";
+const JSON_TYPE_NAME: &str = "Json";
+const JSON_B_TYPE_NAME: &str = "JsonB";
+
+const NATIVE_TYPE_CONSTRUCTORS: &[NativeTypeConstructor] = &[
+    NativeTypeConstructor::without_args(SMALL_INT_TYPE_NAME, &[ScalarType::Int]),
+    NativeTypeConstructor::without_args(INTEGER_TYPE_NAME, &[ScalarType::Int]),
+    NativeTypeConstructor::without_args(BIG_INT_TYPE_NAME, &[ScalarType::BigInt]),
+    NativeTypeConstructor::with_optional_args(DECIMAL_TYPE_NAME, 2, &[ScalarType::Decimal]),
+    NativeTypeConstructor::without_args(INET_TYPE_NAME, &[ScalarType::String]),
+    NativeTypeConstructor::without_args(CITEXT_TYPE_NAME, &[ScalarType::String]),
+    NativeTypeConstructor::without_args(OID_TYPE_NAME, &[ScalarType::Int]),
+    NativeTypeConstructor::without_args(REAL_TYPE_NAME, &[ScalarType::Float]),
+    NativeTypeConstructor::without_args(DOUBLE_PRECISION_TYPE_NAME, &[ScalarType::Float]),
+    NativeTypeConstructor::with_optional_args(VARCHAR_TYPE_NAME, 1, &[ScalarType::String]),
+    NativeTypeConstructor::with_optional_args(CHAR_TYPE_NAME, 1, &[ScalarType::String]),
+    NativeTypeConstructor::without_args(TEXT_TYPE_NAME, &[ScalarType::String]),
+    NativeTypeConstructor::without_args(BYTE_A_TYPE_NAME, &[ScalarType::Bytes]),
+    NativeTypeConstructor::with_optional_args(TIMESTAMP_TYPE_NAME, 1, &[ScalarType::DateTime]),
+    NativeTypeConstructor::with_optional_args(TIMESTAMP_TZ_TYPE_NAME, 1, &[ScalarType::DateTime]),
+    NativeTypeConstructor::without_args(DATE_TYPE_NAME, &[ScalarType::DateTime]),
+    NativeTypeConstructor::with_optional_args(TIME_TYPE_NAME, 1, &[ScalarType::DateTime]),
+    NativeTypeConstructor::with_optional_args(TIME_TZ_TYPE_NAME, 1, &[ScalarType::DateTime]),
+    NativeTypeConstructor::without_args(BOOLEAN_TYPE_NAME, &[ScalarType::Boolean]),
+    NativeTypeConstructor::with_optional_args(BIT_TYPE_NAME, 1, &[ScalarType::String]),
+    NativeTypeConstructor::with_optional_args(VAR_BIT_TYPE_NAME, 1, &[ScalarType::String]),
+    NativeTypeConstructor::without_args(UUID_TYPE_NAME, &[ScalarType::String]),
+    NativeTypeConstructor::without_args(JSON_TYPE_NAME, &[ScalarType::Json]),
+    NativeTypeConstructor::without_args(JSON_B_TYPE_NAME, &[ScalarType::Json]),
+];
+
+const CONSTRAINT_SCOPES: &[ConstraintScope] = &[
+    ConstraintScope::GlobalPrimaryKeyKeyIndex,
```

Comment:
https://github.com/prisma/introspection-ci/blob/feat/cockroachdb/introspection-analysis/output/cockroachdb-serverless/system_stderr.log

Looks like primary keys can have the same name on different models on cockroach. Then the constraint scope `GlobalPrimaryKeyKeyIndex` here would be incorrect.